games-simulation/simutrans/simutrans-120.2.1.recipe

```diff
@@ -0,0 +1,86 @@
+SUMMARY="A transport simulation game"
+DESCRIPTION="Simutrans is a freeware and open-source transportation \
+simulator. Your goal is to establish a successful transport company. \
+Transport passengers, mail and goods by rail, road, ship, and even air. \
+Interconnect districts, cities, public buildings, industries and \
+tourist attractions by building a transport network you always dreamed \
+of."
+HOMEPAGE="http://www.simutrans.com"
+SvnRevision="8121"
+GitRevision="83664a991f819677515ec6bb14e2cf5ce126d8ba"
+SOURCE_URI="https://github.com/aburch/simutrans/archive/$GitRevision.tar.gz"
+CHECKSUM_SHA256="4fb73683e648ddc09964a7e4d58f7928b94a9d28088c68635ae9a20789b39c43"
+SOURCE_FILENAME="simutrans-$portVersion.tar.gz"
+SOURCE_DIR="simutrans-$GitRevision"
+SOURCE_URI_2="https://downloads.sourceforge.net/project/simutrans/simutrans/120-2-1/simutrans-src-120-2-1.zip"
+CHECKSUM_SHA256_2="7dfbd1c44f47ddd8caedb3da77dac8262652f96e1c9ffd890b67757e08ddc942"
+SOURCE_FILENAME_2="simutrans-sf-$portVersion.zip"
+SOURCE_DIR_2=""
+COPYRIGHT="1997-2004 Hj. Malthaner
+	2005-2017 The Simutrans Team"
+LICENSE="Artistic"
+REVISION="1"
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+PROVIDES="
+	simutrans$secondaryArchSuffix = $portVersion
+	app:simutrans$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libsdl$secondaryArchSuffix
+	lib:libsdl_mixer$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libpng$secondaryArchSuffix
+	lib:libbz2$secondaryArchSuffix
+	simutrans_pakset$secondaryArchSuffix >= 120.2
+	timgmsoundfont
+	"
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libsdl$secondaryArchSuffix
+	devel:libsdl_mixer$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:libpng$secondaryArchSuffix
+	devel:libbz2$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:g++$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	cmd:autoreconf
+	cmd:aclocal
+	cmd:make
+	cmd:ld$secondaryArchSuffix
+	cmd:strip$secondaryArchSuffix
+	cmd:sdl_config$secondaryArchSuffix
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+GLOBAL_WRITABLE_FILES="
+	non-packaged/$relativeAppsDir/simutrans/config directory keep-old
+	"
+
+BUILD()
+{
+	autoreconf configure.ac
+	export CFLAGS+=-DREVISION=$SvnRevision
+	runConfigure ./configure
+	make $jobArgs
+	strip sim
+}
+
+INSTALL()
+{
+	fixPkgconfig
+	cp $sourceDir2/simutrans/text/*.tab simutrans/text/
+	cp sim simutrans/simutrans
+	mkdir -p $appsDir/simutrans
+	mkdir -p $prefix/non-packaged/$relativeAppsDir/simutrans/config
```

Comment:
Actually yes, it is possible (I checked it several times for the previous recipes with a freshly installed system - took me some time to find a working solution). The config files must be writable - simutrans can't properly run if they're not. In addition, there must be an initial config file - the one from the package - otherwise simutrans won't start properly. The package includes a symlink in $appsDir/simutrans/config to the writable folder in non-packaged. When installing a simutrans_pak package for the first time a configuration file from the package is copied to non-packaged in the corresponding directory (I think I got the idea from the python package). When you run simutrans, settings like preferred language are saved to the file (if you can't write them simutrans shows an error message or might even crash). If you update the simutrans/simutrans_pak package the initial config file is not copied if there already is a config file (rule "keep-old" for the GLOBAL_WRITABLE_FILES).
The reason for not using $settingsDir was probably that - at the time of the very first simutrans recipe - more people on IRC were in favour of non-packaged or no one cared enough for one or the other (I didn't have any preferences of my own). Traditionally, even in Linux, simutrans generates a "Simutrans" folder rather than a ".simutrans" folder in the home directory that includes some config files, but also savegames or screenshots that a user might search for in order to share or upload them. The config files from the paks can be found in the apps directory. I thought it was a good idea to keep both the location of the paks config files and the Simutrans folder because people could just follow the "official" simutrans manual.
If you insist on a simutrans folder in the settings directory, is there a way to inform users during or after the installation that screenshots etc. can be found there instead of the Simutrans folder in the home directory as on the other platforms? I'll have to add a patchfile so that simutrans won't generate the home/Simutrans directory when saving games, etc - that will take some time to find in the simutrans source code.
If you don't mind the home/Simutrans folder I'll just change the symlink for the pak's config in the apps directory from non-packaged to the settings directory ($settingsDir/simutrans/config instead of $prefix/non-packaged/$relativeAppsDir/simutrans/config). Then the simutrans manual would still work for haiku - as long as I don't have to remove the symlink.